### PR TITLE
Add recipe for Edward

### PR DESCRIPTION
**Pull request by souravsingh**

### CODE REVIEW

File: recipes/edward/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  description: |
+    Edward is a Python library for probabilistic modelling,
+    inference and criticism. Edward is built on top of Tensorflow.
+  doc_url: http://edwardlib.org/
+  dev_url: https://github.com/blei-lab/edward
+
+extra:
+  recipe-maintainers:
+    - souravsingh
+    - dustinvtran
```
Comment (synapticarbors) [inline recipes/edward/meta.yaml]: @dustinvtran -- are you ok being added as a co-maintainer of the `edward` recipe on conda-forge?

File: recipes/edward/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - edward
+    - edward.criticisms
+    - edward.inferences
+    - edward.inferences.conjugacy
+    - edward.models
+    - edward.util
+
+about:
+  home: http://edwardlib.org
+  license: Apache
```
Comment (synapticarbors) [inline recipes/edward/meta.yaml]: `license: Apache-2.0`

File: recipes/edward/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  license_file: LICENSE.txt
```
Comment (synapticarbors) [inline recipes/edward/meta.yaml]: The license file isn't packaged in the source distribution due to a missing `MANIFEST.in`. Please manually package the license and raise an issue or PR (put url in a comment in the recipe for the issue):

https://conda-forge.org/docs/meta.html#packaging-the-license-manually

File: recipes/edward/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - numpy >=1.7
```
Comment (synapticarbors) [inline recipes/edward/meta.yaml]: I don't think `numpy` or `six` is required for the `build` as the recipe is using `--single-version-externally-managed`

File: recipes/edward/meta.yaml (hunk 1 above)
Comment (dustinvtran) [inline recipes/edward/meta.yaml]: yep, that's fine.

File: recipes/edward/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    - edward.criticisms
+    - edward.inferences
+    - edward.inferences.conjugacy
+    - edward.models
+    - edward.util
+
+about:
+  home: http://edwardlib.org
+  license: Apache-2.0
+  license_file: '{{ environ["edward"] }}/LICENSE.txt'
```
Comment (synapticarbors) [inline recipes/edward/meta.yaml]: The `license_file` line is incorrect. It should be:

```
license_file: '{{ environ["RECIPE_DIR"] }}/LICENSE'
```

**Comment by synapticarbors** [comment]
@souravsingh I think once you fix the issue that is causing this to fail (which I describe in a different comment), this should be good to merge.

File: recipes/edward/meta.yaml (hunk 5 above)
Comment (souravsingh) [reply recipes/edward/meta.yaml]: My bad, I thought ``RECIPE_DIR`` is a placeholder where you will replace it with your variable according to your recipe.

**Comment by synapticarbors** [comment]
Somehow I totally missed that you didn't copy the license into the `recipes/edward` directory in this PR. You need to copy it (https://github.com/blei-lab/edward/blob/master/LICENSE.txt) along side of the `meta.yaml`. I would name it `LICENSE.txt` to match the upstream repo and then change the line in the `meta.yaml` to match.

**Comment by dougalsutherland** [comment]
Just to put a note here: this is the same problem as https://github.com/conda-forge/tensorflow-feedstock/issues/19. 

The keras recipe can luckily use the theano backend rather than tensorflow for testing on Linux. Edward doesn't have that option. So being able to build and test Edward on conda-forge's infrastructure is probably going to depend on https://github.com/conda-forge/tensorflow-feedstock/issues/6.

Another issue is that Edward absolutely does depend on tensorflow being installed (e.g. [here](https://github.com/blei-lab/edward/blob/59b97249c72d8f93cb55db6ddeefed95f2e74c17/edward/__init__.py#L26) in `edward.__init__`), but doesn't specify an explicit tensorflow dependency because of the gross tensorflow / tensorflow-gpu split. Resolving that so the conda-forge package can properly depend on tensorflow might also need some work.

**Pull request closed without merging**